structure/rbtree/rbtree.go

```diff
@@ -0,0 +1,457 @@
+// Red-Black Tree is a kind of self-balancing binary search tree.
+// Each node stores "color" ("red" or "black"), used to ensure that the tree remains balanced during insertions and deletions.
+//
+// For more details check out those links below here:
+// Programiz article : https://www.programiz.com/dsa/red-black-tree
+// Wikipedia article: https://en.wikipedia.org/wiki/Red_black_tree
+// authors [guuzaa](https://github.com/guuzaa)
+// see rbtree.go
+
+package rbtree
+
+import (
+	"fmt"
+
+	"github.com/TheAlgorithms/Go/constraints"
+)
+
+type Color byte
+
+const (
+	Black Color = iota
+	Red
+)
+
+type Node[T constraints.Ordered] struct {
```

Comment:
Node already exists in `binarytree` package. Please either modify that and use that or create an abstract version of both since having the same thing in multiple places is not a good practice. When contributing, please read whether what you're implementing already exists or not.

Reply:
The `Node ` existing in `binarysearchtree` doesn't use generics, but `Node ` I submitted does. To create an abstract version, we should modify the existing code. I am willing to do that. Can I change the existing code to use generics ? If not, I modify the Node I submitted .